src/lesson-1-9-demo/SignupForm.jsx

```diff
@@ -1,23 +1,165 @@
 import { useEffect, useState } from "react";
 import "./signup-form.css";
 
+const createUser = async (userData) => {
+  try {
+    const response = await fetch("http://localhost:3001/api/users", {
+      headers: {
+        Accept: "application/json",
+        "Content-Type": "application/json",
+      },
+      method: "POST",
+      body: JSON.stringify(userData),
+    });
+
+    if (response.ok) {
+      return response.json();
+    }
+
+    throw Error("Failed to create user");
+  } catch (error) {
+    console.error({ error });
+    throw error;
+  }
+};
+
+const initialFormState = {
```

Comment:
Here we define the form's initial state, which we later use to compare to the form's current state.

This let's us determine if the form has been touched or is dirty.